core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLInputFormat.java

```diff
@@ -0,0 +1,146 @@
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * InputFormat class for XMLReader plugin.
+ */
+public class XMLInputFormat extends FileInputFormat<LongWritable, Map<String, String>> {
+  public static final String XML_INPUTFORMAT_NODE_PATH = "xml.inputformat.node.path";
+  public static final String XML_INPUTFORMAT_PATTERN = "xml.inputformat.pattern";
+  public static final String XML_INPUTFORMAT_REPROCESSING_REQUIRED = "xml.inputformat.reprocessing.required";
+  public static final String XML_INPUTFORMAT_PROCESSED_DATA_TEMP_FILE = "xml.inputformat.processed.data.temp.file";
+  public static final String XML_INPUTFORMAT_FILE_ACTION = "xml.inputformat.file.action";
+  public static final String XML_INPUTFORMAT_TARGET_FOLDER = "xml.inputformat.target.folder";
+
+  @Override
+  public RecordReader<LongWritable, Map<String, String>> createRecordReader(InputSplit split,
+                                                                            TaskAttemptContext context)
+    throws IOException {
+    return new XMLRecordReader((FileSplit) split, context.getConfiguration());
+  }
+
+  protected boolean isSplitable(JobContext context, Path file) {
+    //XML files are not splittable hence returning false.
+    return false;
+  }
+
+  @Override
+  protected List<FileStatus> listStatus(JobContext job) throws IOException {
+    List<FileStatus> listStatus = super.listStatus(job);
+    if (listStatus != null && !listStatus.isEmpty()) {
+      //apply pattern on the file list status.
+      String pattern = job.getConfiguration().get(XML_INPUTFORMAT_PATTERN);
+      if (StringUtils.isNotEmpty(pattern)) {
+        listStatus = getPatternMatchedListStatus(pattern, listStatus);
+      }
+      if (CollectionUtils.isNotEmpty(listStatus)) {
+        listStatus = getPreProcessedFilteredListStatus(job.getConfiguration(), listStatus);
+      }
+    }
+    return listStatus;
+  }
+
+  /**
+   * Method to filter files from the preprocessed file list.
+   * @param conf
+   * @param listStatusList
+   * @return List<FileStatus> - filtered file list.
+   */
+  private List<FileStatus> getPreProcessedFilteredListStatus(Configuration conf, List<FileStatus> listStatusList) {
+    String processingReq = conf.get(XML_INPUTFORMAT_REPROCESSING_REQUIRED);
+    String processedDataTempFile = conf.get(XML_INPUTFORMAT_PROCESSED_DATA_TEMP_FILE);
```

Comment:
When you have multiple Maps running, Is this one per mapper ? Why does user need to specify this ?